woodwork/tests/schema/test_schema.py

```diff
@@ -551,3 +551,59 @@ def test_removes_time_index_via_tags(sample_column_names, sample_inferred_logica
     schema.reset_semantic_tags('signup_date')
     assert schema.semantic_tags['signup_date'] == set()
     assert schema.time_index is None
+
+
+def test_set_index(sample_column_names, sample_inferred_logical_types):
+    schema = Schema(sample_column_names, sample_inferred_logical_types)
+    assert schema.index is None
+    assert schema.semantic_tags['id'] == {'numeric'}
+    assert schema.semantic_tags['age'] == {'numeric'}
+
+    schema.set_index('id')
+    assert schema.index == 'id'
+    assert schema.semantic_tags['id'] == {'index'}
+
+    schema.set_index('age')
+    assert schema.index == 'age'
+    assert schema.semantic_tags['age'] == {'index'}
+    assert schema.semantic_tags['id'] == {'numeric'}
+
+    schema.set_index(None)
+    assert schema.index is None
+    assert schema.semantic_tags['age'] == {'numeric'}
+    assert schema.semantic_tags['id'] == {'numeric'}
+
+
+def test_set_index_errors(sample_column_names, sample_inferred_logical_types):
+    schema = Schema(sample_column_names, sample_inferred_logical_types)
+
+    error = re.escape("Specified index column `testing` not found in Schema.")
+    with pytest.raises(LookupError, match=error):
+        schema.set_index('testing')
+
+
+def test_set_index_twice(sample_column_names, sample_inferred_logical_types):
+    schema = Schema(sample_column_names, sample_inferred_logical_types, index='id')
+
+    original_schema = schema._get_subset_schema(list(schema.columns.keys()))
+
+    schema.set_index(None)
+    assert schema.index is None
```

Comment:
This flow is setting the index to `id` on init, then changing to `None`, then changing back to `id`. Is that what you wanted here? I was originally thinking this was confirming there were no issues with setting the index to the same value twice, but I might have misunderstood.

Reply:
you're totally right! Mistake on my part--thanks for catching.